frontend/ui/data/isolanguage.lua

```diff
@@ -0,0 +1,339 @@
+    vol = _("Volapük"), -- constructed language
+    wln = _("Walloon"),
+    xho = _("Xhosa"),
+    yid = _("Yiddish"), -- macrolanguage
+    yor = _("Yoruba"),
+    zha = _("Zhuang"), -- macrolanguage
+    zho = _("Chinese"), -- macrolanguage
+    zul = _("Zulu"),
+
+    -- second chunk of names are languages present in iso-639-2
+    -- this is only as complete as it needs to be to cover the languages in dictionaries.lua
+    ady = _("Adyghe"),
+    alt = _("Southern Altai"),
+    ang = _("Old English"), -- (ca. 450-1100)
+    arc = _("Official Aramaic"), -- (700-300 BCE)
+    ast = _("Asturian"),
+    bik = _("Bikol"), -- macrolanguage
+    ceb = _("Cebuano"),
+    chr = _("Cherokee"),
+    cop = _("Coptic"),
+    crh = _("Crimean Tatar"),
+    dsb = _("Lower Sorbian"),
+    dum = _("Middle Dutch"), -- (ca. 1050-1350)
+    enm = _("Middle English"), --  (1100-1500)
+    frm = _("Middle French"), -- (ca. 1400-1600)
+    fro = _("Old French"), -- (842-ca. 1400)
+    fur = _("Friulian"),
+    goh = _("Old High German"), -- (ca. 750-1050)
+    got = _("Gothic"),
+    grc = _("Ancient Greek"), -- (to 1453)
+    gsw = _("Swiss German"),
+    haw = _("Hawaiian"),
+    hil = _("Hiligaynon"),
+    kum = _("Kumyk"),
+    lad = _("Ladino"),
+    mnc = _("Manchu"),
+    nds = _("Low German"),
+    non = _("Old Norse"),
+    ota = _("Ottoman Turkish"), -- (1500-1928)
+    rup = _("Macedo-Romanian"),
+    sah = _("Yakut"),
+    scn = _("Sicilian"),
+    sco = _("Scots"),
+    sga = _("Old Irish"), -- (to 900)
+    shn = _("Shan"),
+    sux = _("Sumerian"),
+    syc = _("Classical Syriac"),
+    tpi = _("Tok Pisin"),
+
+    -- third chunk of names are only present in iso639-3 standard
+    -- this is only as complete as it needs to be to cover the languages in dictionaries.lua
+    aii = _("Assyrian Neo-Aramaic"),
+    apc = _("Levantine Arabic"),
+    ary = _("Moroccan Arabic"),
+    bnt = _("Bantu"),
+    cim = _("Cimbrian"),
+    cmn = _("Mandarin Chinese"),
+    dlm = _("Dalmatian"),
+    hrx = _("Hunsrik"),
+    izh = _("Ingrian"),
+    jje = _("Jejueo"),
+    kea = _("Kabuverdianu"),
+    lld = _("Ladin"),
+    lzz = _("Laz"),
+    mch = _("Maquiritari"),
+    nch = _("Central Huasteca Nahuatl"),
+    nci = _("Classical Nahuatl"),
+    nrf = _("Jèrriais"), -- Norman
+    pdc = _("Pennsylvania German"),
+    pdt = _("Plautdietsch"),
+    phl = _("Phalura"),
+    stq = _("Saterfriesisch"), -- Saterland Frisian
+    tft = _("Ternate"),
+    txb = _("Tokharian B"),
+    txg = _("Tangut"),
+    vec = _("Venetian"),
+    vep = _("Veps"),
+    wym = _("Wymysorys"), -- Vilamovian
+    xcl = _("Classical Armenian"),
+    yol = _("Yola"),
+    yue = _("Yue Chinese"),
+
+    -- these are made up language codes for very specific languages which do not map to a specific ISO3 code
+    xxpf = _("Proto Finnic"),
+    xxpg = _("Proto Germanic"),
+    xxpw = _("Proto West-Germanic"),
+    xxps = _("Proto Sami"),
+    xxsl = _("Proto Slavic"),
```

Comment:
They're generally spelled more like this, cf. <https://www.britannica.com/topic/Indo-European-languages/Characteristic-developments-of-Indo-European-languages#ref603329>.
```suggestion
    xxpf = _("Proto-Finnic"),
    xxpg = _("Proto-Germanic"),
    xxpw = _("Proto-West Germanic"),
    xxps = _("Proto-Sami"),
    xxsl = _("Proto-Slavic"),
```

Though I'll agree that Proto-West Germanic looks a bit odd. ^_^